### ISSUE TITLE
Does Glab support authenticate Gitlab using HTTP protocol?

### ISSUE THREAD

**Issue by logbasex** [profclems/glab#640]
**Describe the feature or problem you'd like to solve**
My company hosted GitLab repositories at a custom domain `gitlab.companyname.io` using `HTTP` protocol and when I trying to authenticate by running `glab auth login`, I don't see any `HTTP` protocol option (just only `HTTPS` and `SSH` protocol). Therefore I always get 401 HTTP response status code.

**Comment by profclems** [comment]
Hi @logbasex, I would recommend using git over `ssh` for domains using `HTTP` protocol.

To make glab make API request over http, set the `api_protocol` for the host config to `http` and try to login again:
```
$ glab config set api_protocol http --host gitlab.companyname.io

$ glab auth login
```
If the issue still persist, then you'd have to directly specify the token and host with flag and later set the `api_protocol` and `git_protocol` using the `config` command:
```
$ glab auth login -h gitlab.companyname.io -t <your-token-here>

$ glab config set api_protocol http --host gitlab.companyname.io

$ glab config set git_protocol http --host gitlab.companyname.io
```
You can then run `glab auth status` to check the authentication status

**Comment by logbasex** [comment]
None of the above solutions work for me. `glab auth status` show an error message: `gitlab.companyname.io: api call failed: Get "https://gitlab.companyname.io/api/v4/user": http: server gave HTTP response to HTTPS client
`

**Comment by logbasex** [comment]
and thisAPI calls for gitlab.com are made over HTTPS protocol

**Comment by logbasex** [comment]
Details:
```
gitlab.com
  x gitlab.com: api call failed: GET https://gitlab.com/api/v4/user: 401 {message: 401 Unauthorized}
  ✓ Git operations for gitlab.com configured to use ssh protocol.
  ✓ API calls for gitlab.com are made over https protocol
  ✓ REST API Endpoint: https://gitlab.com/api/v4/
  ✓ GraphQL Endpoint: https://gitlab.com/api/graphql/
  x No token provided
companyname.gitlab.io
  x companyname.gitlab.io: api call failed: GET https://gitlab.com/users/sign_in: 503 failed to parse unknown error format
  ✓ Git operations for companyname.gitlab.io configured to use http protocol.
  ✓ API calls for companyname.gitlab.io are made over https protocol
  ✓ REST API Endpoint: https://companyname.gitlab.io/api/v4/
  ✓ GraphQL Endpoint: https://companyname.gitlab.io/api/graphql/
  ✓ Token: ********************
gitlab.companyname.io
  x gitlab.companyname.io: api call failed: Get "https://gitlab.companyname.io/api/v4/user": http: server gave HTTP response to HTTPS client
  ✓ Git operations for gitlab.companyname.io configured to use http protocol.
  ✓ API calls for gitlab.companyname.io are made over http protocol
  ✓ REST API Endpoint: http://gitlab.companyname.io/api/v4/
  ✓ GraphQL Endpoint: http://gitlab.companyname.io/api/graphql/
  ✓ Token: ********************

```

**Comment by profclems** [comment]
There was a little issue with the api client using `api_protocol`... It should be fixed by #641. 
Version 1.16.0 is due today and I will do the release soon

Thanks for reporting this issue @logbasex
